### PR TITLE
dev/alpha example of GCP for all data sources

### PR DESCRIPTION
**Pull request by eschultink**
### Features
  - example of using GCP host connecting to all data sources
  - improvements to init stuff

### Change implications

 - dependencies added/changed? **no**

### CODE REVIEW

File: infra/examples-dev/gcp/README.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
+# example-google-workspace
+** alpha quality example atm **
+
+A Terraform root module to provision GCP project for Psoxy, configure it, and create necessary infra
+to for connections to all supported Google Workspace sources, with state stored to local filesystem.
```
Comment (jlorper) [inline infra/examples-dev/gcp/README.md:5]:
```suggestion
for connections to all supported Google Workspace sources, with state stored to local filesystem.
```

File: infra/examples-dev/gcp/README.md [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+## Usage
+
+### Prereqs
+
+You'll need:
+- a Bash-like shell environment on Linux, MacOS, or [WSL on Windows](https://learn.microsoft.com/en-us/windows/wsl/install).
+- [`git` installed](https://git-scm.com/book/en/v2/Getting-Started-Installing-Git), although it
+  is usually included on those platforms (check with `git --version`).
+- an AWS account and credentials, as described in [Psoxy's AWS - Getting Started docs](https://github.com/Worklytics/psoxy/blob/v0.4.17/docs/aws/getting-started.md)
```
Comment (jlorper) [inline infra/examples-dev/gcp/README.md:19]: point links to `main` so they go to latest version always?
```suggestion
- an AWS account and credentials, as described in [Psoxy's AWS - Getting Started docs](https://github.com/Worklytics/psoxy/blob/main/docs/aws/getting-started.md)
```

Comment (jlorper) [reply infra/examples-dev/gcp/README.md:19]: why AWS is needed at all? Isn't this the GCP example?

Comment (jlorper) [reply infra/examples-dev/gcp/README.md:19]: Also, the GCP project is not created, it should exist, right? Shouldn't it be listed as a pre-req too?

File: infra/examples-dev/gcp/variables.tf [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+    "us-central1",
+    "us-west1",
+  ]
+}
+
+variable "enabled_connectors" {
+  type        = list(string)
+  description = "list of ids of connectors to enabled; see modules/worklytics-connector-specs"
+
+  default = [
```
Comment (jlorper) [inline infra/examples-dev/gcp/variables.tf:74]: should this be empty by default, to force the customer to fill it in?

**Review by jlorper — APPROVED** [review]
A couple of suggestions

File: infra/examples-dev/gcp/README.md (hunk 2 above)
Comment (eschultink) [reply infra/examples-dev/gcp/README.md:19]: well, imho it's like an interface - if we move stuff around the link will break

re: AWS, yes should be changed to gcp

File: infra/examples-dev/gcp/variables.tf (hunk 3 above)
Comment (eschultink) [reply infra/examples-dev/gcp/variables.tf]: yeah, that's fair; we should force explicit choice at least.

**Pull request merged**